### PR TITLE
Feat: ERC-20 token list in the account details page

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/composables/fungibleTokens.ts [hunk 1]
```diff
@@ -66,16 +68,16 @@ export function useFungibleTokens() {
   }
 
   async function loadAvailableTokens() {
-    const response: IToken[] = camelCaseKeysDeep(await fetchAllPages(
-      () => fetchFromMiddleware('/v2/aex9?by=name&limit=100&direction=forward'),
-      fetchFromMiddleware,
-    ));
+    const tokensFetchPromises = protocolsInUse.value.map(
```
Comment (martinkaintas) [inline src/composables/fungibleTokens.ts:71]: I suggest to keep tokens grouped by protocol instead of putting them all in the same array

Comment (peronczyk) [reply src/composables/fungibleTokens.ts:88]: OK, done. Please take a look again @martinkaintas .
Actually this change took me a lot time, but it make sense.

**Review by martinkaintas — CHANGES_REQUESTED** [review]

File: src/types/index.ts [hunk 2]
```diff
@@ -203,7 +231,7 @@ export interface IAccountRaw {
  * Account stored on the application store.
  */
 export interface IAccount extends IHdWalletAccount, IAccountRaw {
-  address: Encoded.AccountAddress; // TODO use `string` as we use not only AE protocol
+  address: Encoded.AccountAddress;
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/types/index.ts:234]: I started to do this but it required too much changes within this PR that was actually not related to the scope. I created separate task for this: https://github.com/superhero-com/superhero-wallet/issues/2570

File: src/composables/fungibleTokens.ts [hunk 3]
```diff
@@ -33,99 +38,88 @@ type ContractInitializeOptions = Omit<Parameters<typeof Contract.initialize>[0],
  * List of all custom tokens available (currently only AE network).
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/composables/fungibleTokens.ts:38]: You're right - it's not only about the AE tokens anymore. I'll update the comment.

File: src/composables/fungibleTokens.ts [hunk 4]
```diff
@@ -33,99 +38,88 @@ type ContractInitializeOptions = Omit<Parameters<typeof Contract.initialize>[0],
...
-      accumulator[token.contractId] = token;
+    tokensAvailable.value = tokens.reduce((accumulator, token) => {
+      const { contractId, protocol } = token;
+      if (!accumulator[protocol]) {
+        accumulator[protocol] = {} as AssetList;
+      }
+      accumulator[protocol]![contractId] = token;
       return accumulator;
-    }, {} as ITokenList);
+    }, {} as typeof tokensAvailable.value);
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/composables/fungibleTokens.ts:104]: I don't see much difference. In your example you are still flatening the results for the reason of counting them. And then you needed to add another map. In my code I don't need additional map.

**Review by martinkaintas — APPROVED** [review]

**Review by CedrikNikita — APPROVED** [review]
Code changes look reasonable.

**Comment by Liubov-crypto** [comment]
LGTM

**Pull request merged**